airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/message/MessageQueue.kt

```diff
@@ -29,6 +31,8 @@ abstract class ChannelMessageQueue<T> : MessageQueue<T> {
     override suspend fun close() {
         channel.close()
     }
+    @OptIn(DelicateCoroutinesApi::class)
+    override fun isClosedForPublish(): Boolean = channel.isClosedForSend
```

Comment:
Another option besides opting in is to add an explicit closed flag to the interface